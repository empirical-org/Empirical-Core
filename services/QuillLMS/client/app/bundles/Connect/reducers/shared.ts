
export function receiveQuestionsData(currentstate, action) {
  if (action.language) {
    return Object.assign({}, currentstate, {
      translated_data: {
        ...currentstate.translated_data,
        [action.language] : action.data
      },
    });
  } else {
    return Object.assign({}, currentstate, {
      hasreceiveddata: true,
      data: action.data,
    });
  }
}

export function receiveQuestionData(currentstate, action) {
  return Object.assign({}, currentstate, {
    data: Object.assign({}, currentstate.data, {
      [action.uid]: action.data,
    })
  })
}
