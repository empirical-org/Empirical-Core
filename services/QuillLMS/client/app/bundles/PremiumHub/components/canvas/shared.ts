export interface CanvasInstance {
  client_id: string;
  id: number;
  client_secret: string;
  school_ids: number[];
  school_names: string[];
  url: string;
}

export interface School {
  id: number;
  name: string;
  subscription_status: any;
  subscriptions: any[];
}
