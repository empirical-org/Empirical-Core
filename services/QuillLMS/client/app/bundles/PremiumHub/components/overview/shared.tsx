export const USAGE_HIGHLIGHTS = 'Usage Highlights'
export const PROFESSIONAL_DEVELOPMENT_AND_SUPPORT = 'Professional Development and Support'
export const ACCOUNT_MANAGEMENT = 'Account Management'
export const PREMIUM_REPORTS = 'Premium Reports'
export const INTEGRATIONS = 'Integrations'

export const iconLinkBase = `${process.env.CDN_URL}/images/pages/administrator/overview`

export const SECTION_NAME_TO_ICON_URL = {
  [USAGE_HIGHLIGHTS]: `${iconLinkBase}/bulb.svg`,
  [PROFESSIONAL_DEVELOPMENT_AND_SUPPORT]: `${iconLinkBase}/students.svg`,
  [ACCOUNT_MANAGEMENT]: `${iconLinkBase}/pencil.svg`,
  [PREMIUM_REPORTS]: `${iconLinkBase}/bar-graph-increasing-black.svg`,
  [INTEGRATIONS]: `${iconLinkBase}/checkbox-multiple.svg`
}
