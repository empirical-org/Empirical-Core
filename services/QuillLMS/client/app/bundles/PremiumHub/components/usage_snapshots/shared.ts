export const SMALL = 'small'
export const MEDIUM = 'medium'
export const POSITIVE = 'positive'
export const NEGATIVE = 'negative'
