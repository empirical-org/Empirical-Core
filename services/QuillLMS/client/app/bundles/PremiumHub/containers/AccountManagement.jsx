import React from 'react';

const AccountManagement = ({ adminId, accessType, }) => {

  return (
    <div className="white-background-accommodate-footer premium-hub-overview">
      <div className="container">
        <h1>Account Management</h1>
      </div>
    </div>
  );
}

export default AccountManagement
