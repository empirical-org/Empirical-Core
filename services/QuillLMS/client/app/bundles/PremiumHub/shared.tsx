import * as React from 'react'
import * as _ from 'lodash'

import * as md5 from 'md5'

import { unorderedArraysAreEqual, } from '../../modules/unorderedArraysAreEqual'

export const RESTRICTED = 'restricted'
export const LIMITED = 'limited'
export const FULL = 'full'
export const LOADING = 'loading'

export const APPROVED = 'Approved'
export const PENDING = 'Pending'
export const DENIED = 'Denied'
export const SKIPPED = 'Skipped'

export const OVERVIEW = 'overview'
export const SKILL = 'skill'
export const STUDENT = 'student'

export const DIAGNOSTIC_REPORT_DEFAULT_CELL_WIDTH = '182px'
export const groupByDropdownOptions = [{ label: 'Grade', value: 'grade' }, { label: 'Teacher', value: 'teacher' }, { label: 'Classroom', value: 'classroom' }]
export const diagnosticTypeDropdownOptions = [
  { label: 'Starter Baseline Diagnostic (Pre)', value: 'Starter Baseline Diagnostic (Pre)' },
  { label: 'Starter Growth Diagnostic (Post)', value: 'Starter Growth Diagnostic (Post)' },
  { label: 'Intermediate Baseline Diagnostic (Pre)', value: 'Intermediate Baseline Diagnostic (Pre)' },
  { label: 'Intermediate Growth Diagnostic (Post)', value: 'Intermediate Growth Diagnostic (Post)' },
  { label: 'Advanced Baseline Diagnostic (Pre)', value: 'Advanced Baseline Diagnostic (Pre)' },
  { label: 'Advanced Growth Diagnostic (Post)', value: 'Advanced Growth Diagnostic (Post)' },
  { label: 'ELL Starter Baseline Diagnostic (Pre)', value: 'ELL Starter Baseline Diagnostic (Pre)' },
  { label: 'ELL Starter Growth Diagnostic (Post)', value: 'ELL Starter Growth Diagnostic (Post)' },
  { label: 'ELL Intermediate Baseline Diagnostic (Pre)', value: 'ELL Intermediate Baseline Diagnostic (Pre)' },
  { label: 'ELL Intermediate Growth Diagnostic (Post)', value: 'ELL Intermediate Growth Diagnostic (Post)' },
  { label: 'ELL Advanced Baseline Diagnostic (Pre)', value: 'ELL Advanced Baseline Diagnostic (Pre)' },
  { label: 'ELL Advanced Growth Diagnostic (Post)', value: 'ELL Advanced Growth Diagnostic (Post)' },
  { label: 'AP Writing Skills Survey', value: 'AP Writing Skills Survey' },
  { label: 'Pre-AP Writing Skills Survey 1', value: 'Pre-AP Writing Skills Survey 1' },
  { label: 'Pre-AP Writing Skills Survey 2', value: 'Pre-AP Writing Skills Survey 2' },
  { label: 'SpringBoard Writing Skills Survey', value: 'SpringBoard Writing Skills Survey' }
]

export const premiumLockImage = <img alt="Gray lock" src={`${process.env.CDN_URL}/images/pages/administrator/premium_lock.svg`} />

export const restrictedElement = (
  <div className="restricted">
    {premiumLockImage}
  </div>
)

export const restrictedPage = (
  <div className="restricted-page">
    {premiumLockImage}
  </div>
)

export function selectionsEqual(firstSelection, secondSelection) {
  return (
    firstSelection == secondSelection || // less strict comparison so that undefined and null are treated as equal
    unorderedArraysAreEqual(firstSelection, secondSelection)
  )
}

export function mapItemsIfNotAll(selectedItems, allItems, mapKey = 'id') {
  // selectedItems may, by design, be a superset of allItems, but if everything in allItems is in selectedItems, we want to treat it as "everything" being selected
  const allItemsSelected = allItems.every((item) => {
    return _.some(selectedItems, item)
  })

  if (allItemsSelected || selectionsEqual(selectedItems, allItems)) return null

  return selectedItems.map(i => i[mapKey])
}

/* Should produce the same output as backend `app/services/payload_hasher.rb:run` */
export function hashPayload(payloadArray: Array<any>) {
  const joinedPayload = payloadArray.join('-')
  return md5(joinedPayload)
}
