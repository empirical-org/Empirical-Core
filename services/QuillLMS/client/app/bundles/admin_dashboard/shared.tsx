import * as React from 'react'

export const RESTRICTED = 'restricted'
export const LIMITED = 'limited'
export const FULL = 'full'

export const premiumLockImage = <img alt="Gray lock" src={`${process.env.CDN_URL}/images/pages/administrator/premium_lock.svg`} />

export const restrictedElement = (
  <div className="restricted">
    {premiumLockImage}
  </div>
)
