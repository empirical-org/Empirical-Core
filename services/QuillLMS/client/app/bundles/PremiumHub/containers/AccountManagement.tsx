import * as React from 'react';
import Pusher from 'pusher-js';

import { requestGet, requestPost, } from '../../../modules/request/index';
import useSnackbarMonitor from '../../Shared/hooks/useSnackbarMonitor';
import { DataTable, DropdownInput, Snackbar, defaultSnackbarTimeout, Spinner, } from '../../Shared/index';
import { ADMIN, PENDING, TEACHER, } from '../../Shared/utils/constants';
import { FULL, RESTRICTED, restrictedElement, RESTRICTED_TEXT, } from '../shared';
import AdminActionModal from '../components/admin_action_modal'
import CreateAndLinkAccountsModal from '../components/create_and_link_accounts_modal'

interface AccountManagementProps {
  accessType: string;
  passedModel: any;
  adminId: number;
}

enum modalNames {
  removeAdminModal = 'removeAdminModal',
  makeAdminModal = 'makeAdminModal',
  createAndLinkAccountsModal = 'createAndLinkAccountsModal'
}

const DEFAULT_MODEL = { teachers: [] }

const teacherColumns = [
  {
    name: 'Name',
    attribute: 'name',
    isSortable: true,
    width: '370px'
  },
  {
    name: 'Role',
    attribute: 'role',
    isSortable: true,
    width: '276px'
  },
  {
    name: 'Students',
    attribute: 'number_of_students',
    isSortable: true,
    width: '156px',
  },
  {
    name: 'Activities Completed',
    attribute: 'number_of_activities_completed',
    isSortable: true,
    width: '156px',
  },
  {
    name: 'Time Spent',
    attribute: 'time_spent',
    isSortable: true,
    width: '156px',
  },
  {
    name: 'Actions',
    attribute: 'actions',
    isActions: true
  }
];

export const AccountManagement: React.SFC<AccountManagementProps> = ({
  accessType,
  passedModel,
  adminId,
}) => {
  const [loading, setLoading] = React.useState(passedModel ? false : true)
  const [model, setModel] = React.useState(passedModel || DEFAULT_MODEL)
  const [showSnackbar, setShowSnackbar] = React.useState(false);
  const [snackbarText, setSnackbarText] = React.useState('');
  const [userIdForModal, setUserIdForModal] = React.useState(null)
  const [showModal, setShowModal] = React.useState(null)
  const [error, setError] = React.useState('')
  const [selectedSchoolId, setSelectedSchoolId] = React.useState(null)

  React.useEffect(getData, [])

  useSnackbarMonitor(showSnackbar, setShowSnackbar, defaultSnackbarTimeout)

  function getData(skipLoading = false) {
    initializePusher(skipLoading);
    requestGet(
      `${process.env.DEFAULT_URL}/admins/${adminId}`,
      (body) => {
        receiveData(body, skipLoading)
      }
    );
  }

  function receiveData(data, skipLoading) {
    if (Object.keys(data).length > 1) {
      setModel(data)
      const defaultSchool = data.schools.find(s => s.id === data.associated_school?.id) || data.schools[0]
      setSelectedSchoolId(defaultSchool?.id)
      setLoading(false)
    } else if (!skipLoading) {
      setModel(data)
      setLoading(true)
    }
  };

  function initializePusher(skipLoading) {
    if (process.env.RAILS_ENV === 'development') {
      Pusher.logToConsole = true;
    }
    const pusher = new Pusher(process.env.PUSHER_KEY, { encrypted: true, });
    const channel = pusher.subscribe(String(adminId));
    channel.bind('admin-users-found', () => {
      getData(skipLoading)
    });
  };

  function addTeacherAccount(data) {
    setError('')
    requestPost(
      `${process.env.DEFAULT_URL}/admins/${adminId}/create_and_link_accounts`,
      data,
      (response) => {
        getData(true)
        setSnackbarText(response.message)
        setShowSnackbar(true)
      },
      (response) => {
        if (response.error) {
          setError(response.error);
        } else {
          // to do, use Sentry to capture error
        }
      }
    );
  };

  function handleClickCreateAndLinkAccountsButton() {
    if (accessType === RESTRICTED) {
      setSnackbarText(RESTRICTED_TEXT)
      setShowSnackbar(true)
    } else {
      setShowModal(modalNames.createAndLinkAccountsModal)
    }
  }

  function handleUserAction(link, data) {
    setError('')
    initializePusher(true)
    requestPost(
      link,
      data,
      (response) => {
        getData(true)
        setSnackbarText(response.message)
        setShowSnackbar(true)
      },
      (response) => {
        if (response.error) {
          setError(response.error);
        } else {
          // to do, use Sentry to capture error
        }
      }
    );
  }

  function onChangeSelectedSchool(selectedSchoolOption) { setSelectedSchoolId(selectedSchoolOption.value) }

  function closeModal() {
    setUserIdForModal(false)
    setShowModal(false)
  }

  function loginAsUser(id) {
    if (accessType === FULL) {
      window.location.href = `/users/${id}/admin_sign_in_classroom_manager`
    } else {
      setSnackbarText(RESTRICTED_TEXT)
      setShowSnackbar(true)
    }
  }

  function viewPremiumReports(id) {
    if (accessType === FULL) {
      window.location.href = `/users/${id}/admin_sign_in_progress_reports`
    } else {
      setSnackbarText(RESTRICTED_TEXT)
      setShowSnackbar(true)
    }
  }

  function resendLoginDetailsForTeacher(id) {
    handleUserAction(`/users/${id}/admin_resend_login_details`, { role: 'teacher', school_id: selectedSchoolId, })
  }

  function unlinkFromSchool(id) {
    handleUserAction(`/users/${id}/admin_unlink_from_school`, { role: 'teacher', })
  }

  function approveAdminRequest(id) {
    handleUserAction(`/users/${id}/approve_admin_request`, { school_id: selectedSchoolId, })
  }

  function denyAdminRequest(id) {
    handleUserAction(`/users/${id}/deny_admin_request`, { school_id: selectedSchoolId, })
  }

  function resendLoginDetailsForAdmin(id) {
    handleUserAction(`/users/${id}/admin_resend_login_details`, { role: 'admin', school_id: selectedSchoolId, })
  }

  function handleConfirmMakeAdmin() {
    handleUserAction(`/users/${userIdForModal}/admin_make_admin`, { school_id: selectedSchoolId, })
    closeModal()
  }

  function handleConfirmRemoveAsAdmin() {
    handleUserAction(`/users/${userIdForModal}/admin_remove_as_admin`, { school_id: selectedSchoolId, })
    closeModal()
  }

  function removeAsAdmin(id) {
    setUserIdForModal(id)
    setShowModal(modalNames.removeAdminModal)
  }

  function makeAdmin(id) {
    setUserIdForModal(id)
    setShowModal(modalNames.makeAdminModal)
  }

  const actionsHash = {
    resendLoginDetailsAdmin: {
      name: 'Resend login details',
      action: (id) => resendLoginDetailsForAdmin(id)
    },
    resendLoginDetailsTeacher: {
      name: 'Resend login details',
      action: (id) => resendLoginDetailsForTeacher(id)
    },
    loginAsAdmin: {
      name: 'Login as admin',
      action: (id) => loginAsUser(id)
    },
    loginAsTeacher: {
      name: 'Login as teacher',
      action: (id) => loginAsUser(id)
    },
    viewPremiumReports: {
      name: 'View premium reports',
      action: (id) => viewPremiumReports(id)
    },
    approveAdminRequest: {
      name: 'Approve admin request',
      action: (id) => approveAdminRequest(id)
    },
    denyAdminRequest: {
      name: 'Deny admin request',
      action: (id) => denyAdminRequest(id)
    },
    removeAsAdmin: {
      name: 'Remove as admin',
      action: (id) => removeAsAdmin(id)
    },
    makeAdmin: {
      name: 'Make admin',
      action: (id) => makeAdmin(id)
    },
    unlinkFromSchool: {
      name: 'Unlink from school',
      action: (id) => unlinkFromSchool(id)
    },
  }

  function actionsForUser(user, relevantSchool) {
    let actions
    if (relevantSchool.role.toLowerCase() === ADMIN) {
      actions = user.last_sign_in ? [] : [actionsHash.resendLoginDetailsAdmin]
      actions = actions.concat([actionsHash.loginAsAdmin, actionsHash.viewPremiumReports, actionsHash.removeAsAdmin])
    } else if (pendingAdminRequest(user, relevantSchool)) {
      actions = user.last_sign_in ? [] : [actionsHash.resendLoginDetailsTeacher]
      actions = actions.concat([actionsHash.loginAsTeacher, actionsHash.viewPremiumReports, actionsHash.approveAdminRequest, actionsHash.denyAdminRequest, actionsHash.unlinkFromSchool])
    } else {
      actions = user.last_sign_in ? [] : [actionsHash.resendLoginDetailsTeacher]
      actions = actions.concat([actionsHash.loginAsTeacher, actionsHash.viewPremiumReports, actionsHash.makeAdmin, actionsHash.unlinkFromSchool])
    }
    return actions
  }

  function pendingAdminRequest(user, relevantSchool) {
    return relevantSchool.role.toLowerCase() === TEACHER && user.admin_info?.approval_status === PENDING && model.admin_approval_requests.find(request => request.admin_info_id === user.admin_info?.id)
  }

  if (loading) {
    return (
      <div className="white-background-accommodate-footer account-management">
        <div className="container">
          <div className="header">
            <h1>Account Management</h1>
            <button className="quill-button contained medium focus-on-light primary" type="button">Create and link accounts</button>
          </div>
          <Spinner />
        </div>
      </div>
    )
  }


  const schoolOptions = model.schools.map(school => ({ value: school.id, label: school.name}))

  const filteredData = model.teachers.filter((d: { school: string }) => d.schools.find(s => s.id === selectedSchoolId)).map(user => {
    const relevantSchool = user.schools.find(s => s.id === selectedSchoolId)
    user.actions = actionsForUser(user, relevantSchool)
    const adminRequest = pendingAdminRequest(user, relevantSchool)
    if (adminRequest) {
      user.role = adminRequest.request_made_during_sign_up ? "🛎️ New user requested to become admin" : "🛎️ Teacher requested to become admin"
    } else {
      user.role = relevantSchool.role
    }
    return user
  })

  const userNameForModal = model.teachers.find(user => user.id === userIdForModal)?.name
  const schoolNameForModal = model.schools.find(s => s.id === selectedSchoolId)?.name

  let accountManagementContent = (
    <React.Fragment>
      <DropdownInput
        className="page-school-selector"
        handleChange={onChangeSelectedSchool}
        isSearchable={true}
        options={schoolOptions}
        value={schoolOptions.find(so => so.value === selectedSchoolId)}
      />
      <div className="data-table-wrapper">
        <DataTable
          className='reporting-format'
          defaultSortAttribute="role"
          headers={teacherColumns}
          rows={filteredData}
          showActions={true}
        />
        <p className="warning-section">
          <span className="warning">Warning:</span> Any changes you make when you
          access a teacher account will impact the teacher and student facing
          dashboards. This list provides you with the ability to sign in to all of the
          teacher accounts for the schools you have admin access.&nbsp;
          <strong>The data above represents usage from this school year, beginning July 1st.</strong>
        </p>
      </div>
    </React.Fragment>
  )

  if (accessType === RESTRICTED) {
    accountManagementContent = restrictedElement
  }

  return (
    <div className="white-background-accommodate-footer account-management">
      <div className="container">
        <Snackbar text={snackbarText} visible={showSnackbar} />
        {showModal === modalNames.makeAdminModal && (
          <AdminActionModal
            bodyText={`Are you sure you want to add ${userNameForModal} as an admin of ${schoolNameForModal}?`}
            handleClickConfirm={handleConfirmMakeAdmin}
            handleCloseModal={closeModal}
            headerText="Add admin?"
          />
        )}
        {showModal === modalNames.removeAdminModal && (
          <AdminActionModal
            bodyText={`Are you sure you want to remove ${userNameForModal} as an admin of ${schoolNameForModal}?`}
            handleClickConfirm={handleConfirmRemoveAsAdmin}
            handleCloseModal={closeModal}
            headerText="Remove admin?"
          />
        )}
        {showModal === modalNames.createAndLinkAccountsModal && (
          <CreateAndLinkAccountsModal
            addTeacherAccount={addTeacherAccount}
            handleCloseModal={closeModal}
            schoolOptions={schoolOptions}
          />
        )}
        <div className="header">
          <h1>Account Management</h1>
          <button className="quill-button contained medium focus-on-light primary" onClick={handleClickCreateAndLinkAccountsButton} type="button">Create and link accounts</button>
        </div>
        {accountManagementContent}
      </div>
    </div>
  );
}

export default AccountManagement
