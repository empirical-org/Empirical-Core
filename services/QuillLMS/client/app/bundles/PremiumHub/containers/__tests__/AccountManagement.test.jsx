import * as React from "react";
import { render, } from "@testing-library/react";

import AccountManagement from '../AccountManagement';
import { FULL, RESTRICTED, LIMITED, } from '../../shared'

const sharedProps = {
  passedModel: { teachers: [], schools: [], admin_approval_requests: [] },
  adminId: 7,
}

describe('AccountManagement container', () => {
  test('should render with a full access type', () => {
    const { asFragment } = render(<AccountManagement {...sharedProps} accessType={FULL} />);
    expect(asFragment()).toMatchSnapshot();
  })

  test('should render with a limited access type', () => {
    const { asFragment } = render(<AccountManagement {...sharedProps} accessType={LIMITED} />);
    expect(asFragment()).toMatchSnapshot();
  })

  test('should render with a restricted access type', () => {
    const { asFragment } = render(<AccountManagement {...sharedProps} accessType={RESTRICTED} />);
    expect(asFragment()).toMatchSnapshot();
  })

});
