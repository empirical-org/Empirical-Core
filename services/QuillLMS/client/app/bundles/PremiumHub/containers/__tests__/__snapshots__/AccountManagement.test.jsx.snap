// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountManagement container should render with a full access type 1`] = `
<DocumentFragment>
  <div
    class="white-background account-management"
  >
    <div
      class="container"
    >
      <div
        class="quill-snackbar "
      />
      <div
        class="header"
      >
        <h1>
          Account Management
        </h1>
        <button
          class="quill-button contained medium focus-on-light primary"
          type="button"
        >
          Create and link accounts
        </button>
      </div>
      <div
        class="dropdown-container input-container inactive   page-school-selector searchable "
        role="button"
        tabindex="0"
      >
        <label />
        <div
          class="dropdown css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-2-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="dropdown__control css-13cymwt-control"
          >
            <div
              class="dropdown__value-container css-1fdsijx-ValueContainer"
            >
              <div
                class="dropdown__placeholder css-1jqq78o-placeholder"
                id="react-select-2-placeholder"
              />
              <div
                class="dropdown__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-2-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="dropdown__input"
                  id="react-select-2-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="dropdown__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="dropdown__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="dropdown__indicator dropdown__dropdown-indicator css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="data-table-wrapper"
      >
        <table
          class="data-table reporting-format"
        >
          <thead>
            <tr
              class="data-table-headers"
            >
              <th
                class="data-table-header undefined sortable"
                scope="col"
                style="width: 394px; min-width: 394px; text-align: right;"
              >
                <button
                  class="interactive-wrapper focus-on-light"
                  type="button"
                >
                  <span>
                    Name
                  </span>
                  <div
                    class="sort-tooltip-container"
                  />
                </button>
              </th>
              <th
                class="data-table-header undefined sortable"
                scope="col"
                style="width: 180px; min-width: 180px; text-align: right;"
              >
                <button
                  class="interactive-wrapper focus-on-light"
                  type="button"
                >
                  <span>
                    Role
                  </span>
                  <div
                    class="sort-tooltip-container"
                  >
                    <img
                      alt="arrow"
                      class="sort-arrow asc"
                      src="https://assets.quill.org/images/icons/icons-arrow.svg"
                    />
                  </div>
                </button>
              </th>
              <th
                class="data-table-header undefined sortable"
                scope="col"
                style="width: 180px; min-width: 180px; text-align: right;"
              >
                <button
                  class="interactive-wrapper focus-on-light"
                  type="button"
                >
                  <span>
                    Students
                  </span>
                  <div
                    class="sort-tooltip-container"
                  />
                </button>
              </th>
              <th
                class="data-table-header undefined sortable"
                scope="col"
                style="width: 180px; min-width: 180px; text-align: right;"
              >
                <button
                  class="interactive-wrapper focus-on-light"
                  type="button"
                >
                  <span>
                    Activities Completed
                  </span>
                  <div
                    class="sort-tooltip-container"
                  />
                </button>
              </th>
              <th
                class="data-table-header undefined sortable"
                scope="col"
                style="width: 180px; min-width: 180px; text-align: right;"
              >
                <button
                  class="interactive-wrapper focus-on-light"
                  type="button"
                >
                  <span>
                    Time Spent
                  </span>
                  <div
                    class="sort-tooltip-container"
                  />
                </button>
              </th>
              <th
                class="data-table-header actions-header"
                scope="col"
              >
                Actions
              </th>
            </tr>
          </thead>
          <tbody
            class="data-table-body"
          />
        </table>
        <p
          class="warning-section"
        >
          <span
            class="warning"
          >
            Warning:
          </span>
           Any changes you make when you access a teacher account will impact the teacher and student facing dashboards. This list provides you with the ability to sign in to all of the teacher accounts for the schools you have admin access.Â 
          <strong>
            The data above represents usage from this school year, beginning July 1st.
          </strong>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AccountManagement container should render with a limited access type 1`] = `
<DocumentFragment>
  <div
    class="white-background account-management"
  >
    <div
      class="container"
    >
      <div
        class="quill-snackbar "
      />
      <div
        class="header"
      >
        <h1>
          Account Management
        </h1>
        <button
          class="quill-button contained medium focus-on-light primary"
          type="button"
        >
          Create and link accounts
        </button>
      </div>
      <div
        class="dropdown-container input-container inactive   page-school-selector searchable "
        role="button"
        tabindex="0"
      >
        <label />
        <div
          class="dropdown css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-3-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="dropdown__control css-13cymwt-control"
          >
            <div
              class="dropdown__value-container css-1fdsijx-ValueContainer"
            >
              <div
                class="dropdown__placeholder css-1jqq78o-placeholder"
                id="react-select-3-placeholder"
              />
              <div
                class="dropdown__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-3-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="dropdown__input"
                  id="react-select-3-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="dropdown__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="dropdown__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="dropdown__indicator dropdown__dropdown-indicator css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="data-table-wrapper"
      >
        <table
          class="data-table reporting-format"
        >
          <thead>
            <tr
              class="data-table-headers"
            >
              <th
                class="data-table-header undefined sortable"
                scope="col"
                style="width: 394px; min-width: 394px; text-align: right;"
              >
                <button
                  class="interactive-wrapper focus-on-light"
                  type="button"
                >
                  <span>
                    Name
                  </span>
                  <div
                    class="sort-tooltip-container"
                  />
                </button>
              </th>
              <th
                class="data-table-header undefined sortable"
                scope="col"
                style="width: 180px; min-width: 180px; text-align: right;"
              >
                <button
                  class="interactive-wrapper focus-on-light"
                  type="button"
                >
                  <span>
                    Role
                  </span>
                  <div
                    class="sort-tooltip-container"
                  >
                    <img
                      alt="arrow"
                      class="sort-arrow asc"
                      src="https://assets.quill.org/images/icons/icons-arrow.svg"
                    />
                  </div>
                </button>
              </th>
              <th
                class="data-table-header undefined sortable"
                scope="col"
                style="width: 180px; min-width: 180px; text-align: right;"
              >
                <button
                  class="interactive-wrapper focus-on-light"
                  type="button"
                >
                  <span>
                    Students
                  </span>
                  <div
                    class="sort-tooltip-container"
                  />
                </button>
              </th>
              <th
                class="data-table-header undefined sortable"
                scope="col"
                style="width: 180px; min-width: 180px; text-align: right;"
              >
                <button
                  class="interactive-wrapper focus-on-light"
                  type="button"
                >
                  <span>
                    Activities Completed
                  </span>
                  <div
                    class="sort-tooltip-container"
                  />
                </button>
              </th>
              <th
                class="data-table-header undefined sortable"
                scope="col"
                style="width: 180px; min-width: 180px; text-align: right;"
              >
                <button
                  class="interactive-wrapper focus-on-light"
                  type="button"
                >
                  <span>
                    Time Spent
                  </span>
                  <div
                    class="sort-tooltip-container"
                  />
                </button>
              </th>
              <th
                class="data-table-header actions-header"
                scope="col"
              >
                Actions
              </th>
            </tr>
          </thead>
          <tbody
            class="data-table-body"
          />
        </table>
        <p
          class="warning-section"
        >
          <span
            class="warning"
          >
            Warning:
          </span>
           Any changes you make when you access a teacher account will impact the teacher and student facing dashboards. This list provides you with the ability to sign in to all of the teacher accounts for the schools you have admin access.Â 
          <strong>
            The data above represents usage from this school year, beginning July 1st.
          </strong>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AccountManagement container should render with a restricted access type 1`] = `
<DocumentFragment>
  <div
    class="white-background account-management"
  >
    <div
      class="container"
    >
      <div
        class="quill-snackbar "
      />
      <div
        class="header"
      >
        <h1>
          Account Management
        </h1>
        <button
          class="quill-button contained medium focus-on-light primary"
          type="button"
        >
          Create and link accounts
        </button>
      </div>
      <div
        class="restricted"
      >
        <img
          alt="Gray lock"
          src="undefined/images/pages/administrator/premium_lock.svg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
